### PR TITLE
fix(surrogates): noise_prior and noise_constraint on SingleTaskGPSurr…

### LINKED ISSUE #762: [Bug]: SingleTaskGPSurrogate: noise_prior and noise_constraint have no effect on GP fitting
### What happened?

When a user specifies a custom noise_prior on SingleTaskGPSurrogate — for example to encode domain knowledge that observation noise is expected to be large — the prior has no effect on the fitted model. The GP always behaves as if the BoTorch default GammaPrior(1.1, 0.05) were used, regardless of what is set in the data model. Concretely, two strategies configured with very different noise priors (e.g. LogNormalPrior(loc=-4, scale=1) vs GammaPrior(concentration=1.1, rate=0.001)) produce identical fitted noise values and identical predictions after tell(). The same silent failure affects noise_constraint: setting it on the surrogate data model has no influence on the optimised hyperparameters.

### Please provide a minimal, reproducible example of the unexpected behavior.

import torch
import numpy as np
from bofire.benchmarks.single import Himmelblau
from bofire.data_models.strategies.api import RandomStrategy as RandomStrategyDataModel
from bofire.strategies.api import RandomStrategy, SoboStrategy
import bofire.data_models.strategies.api as data_models
from bofire.data_models.surrogates.api import BotorchSurrogates, SingleTaskGPSurrogate
from bofire.data_models.priors.api import GammaPrior

torch.manual_seed(42)
np.random.seed(42)

benchmark = Himmelblau()
rs = RandomStrategy(data_model=RandomStrategyDataModel(domain=benchmark.domain))
experiments = benchmark.f(rs.ask(20), return_complete=True)
test_candidates = rs.ask(10)

# Strategy 1: default HVARFNER prior — LogNormal(-4, 1), strongly prefers small noise
s_default = SoboStrategy(data_model=data_models.SoboStrategy(domain=benchmark.domain))
s_default.tell(experiments)

# Strategy 2: extreme large-noise prior — GammaPrior(1.1, 0.001), mode ≈ 100
s_large = SoboStrategy(
    data_model=data_models.SoboStrategy(
        domain=benchmark.domain,
        surrogate_specs=BotorchSurrogates(surrogates=[
            SingleTaskGPSurrogate(
                inputs=benchmark.domain.inputs,
                outputs=benchmark.domain.outputs,
                noise_prior=GammaPrior(concentration=1.1, rate=0.001),
            )
        ]),
    )
)
s_large.tell(experiments)

noise_default = s_default.surrogates.surrogates[0].model.likelihood.noise.item()
noise_large   = s_large.surrogates.surrogates[0].model.likelihood.noise.item()
preds_default = s_default.predict(test_candidates)
preds_large   = s_large.predict(test_candidates)

print(f"Fitted noise (default prior): {noise_default:.6f}")
print(f"Fitted noise (large prior):   {noise_large:.6f}")
print(f"Predictions identical: {np.allclose(preds_default['y_pred'].values, preds_large['y_pred'].values)}")

### Please paste any relevant traceback/logs produced by the example provided.

```shell

```

### BoFire Version

0.3.2.dev45 (local development build from commit acee1667e, dated 2026-05-20).

### Python Version

_No response_

### Operating System

_No response_

### PR DESCRIPTION
**Pull request by R-M-Lee**
…ogate now correctly influence GP fitting

attempt to understand and fix https://github.com/experimental-design/bofire/issues/762

### CODE REVIEW

**Comment by R-M-Lee** [comment]
It looks like the same bug exists in multi_task_gp.py, shape.py, and tanimoto_gp_surrogate.py — all three use the same post-hoc attribute assignment pattern to set the noise prior after model construction. 

Each needs the same fix: construct a GaussianLikelihood with the correct noise_prior and noise_constraint, then pass it to the model constructor via the likelihood parameter.

**Comment by jduerholt** [comment]
@R-M-Lee : I did exaclty the same but with some other changes in addition. I will create a seperate PR, ok?

**Comment by jduerholt** [comment]
Fixed in https://github.com/experimental-design/bofire/pull/766.
